From 695006a9e11849cc2362797bc096ed32ec9fef56 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:41 -0500
Subject: [drm] ast: open key before detect chips

Message-id: <1418852380-13061-31-git-send-email-rclark@redhat.com>
Patchwork-id: 102160
O-Subject: [RHEL7 drm 30/89] drm/ast: open key before detect chips
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit 6064a72983d67dce2a22d78cca21ef0a8d14a8c9

Author:     Y.C. Chen <yc_chen@aspeedtech.com>
AuthorDate: Wed Sep 10 12:07:53 2014 +0800
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:16:27 2014 -0500

    drm/ast: open key before detect chips

    commit 8f372e250ae5f4a5faf87ca1a629d0ae59db65b6 upstream.

    Some config settings like 3rd TX chips will not get correctly
    if the extended reg is protected

    Signed-off-by: Y.C. Chen <yc_chen@aspeedtech.com>
    Reviewed-by: Egbert Eich <eich@suse.de>
    Signed-off-by: Dave Airlie <airlied@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/ast/ast_main.c b/drivers/gpu/drm/ast/ast_main.c
index a2cc6be..b027e33 100644
--- a/drivers/gpu/drm/ast/ast_main.c
+++ b/drivers/gpu/drm/ast/ast_main.c
@@ -67,6 +67,7 @@ static int ast_detect_chip(struct drm_device *dev)
 {
  struct ast_private *ast = dev->dev_private;
  uint32_t data, jreg;
+ ast_open_key(ast);
 
  if (dev->pdev->device == PCI_CHIP_AST1180) {
   ast->chip = AST1100;
-- 
1.7.1